### ISSUE TITLE
[Bug]: Endless import / UI not updated ? 

### ISSUE THREAD

**Issue by Zbuffers** [AssetRipper/AssetRipper#1015]
### Are you on the latest version of AssetRipper?

Yes, I'm on the latest release of AssetRipper.

### Which release are you using?

Windows x64

### Which game did this occur on?

[https://steamdb.info/app/2223740/]

### Which Unity version did this occur on?

2021.3.16f1

### Is the game Mono or IL2Cpp?

Mono

### Describe the issue.

Hi,
This is a strange issue.

Using commit a133532 compiled in release mode
Using options: 
- Scripts: Decompile
- Sprite: Texture
(All others set to default)

When I loaded the game, the import loop run for a very long time (~15+ minutes)
Usually, at the end of the load, the UI refreshes with the treeview of the files, but in this case, it never showed...
The processor core the app was using was still at 100%, (endless loop ?)
So I tried to export the files anyway, thinking that I should get an error, but... no the game seems to have exported nicely !!! ???

Screenshot shows the successfull export, but notice the UI is still reporting the game is loading ???
Strange ;) ...

Kind regards, 
Z

![Untitled](https://github.com/AssetRipper/AssetRipper/assets/7527771/0aba372a-7b8b-478c-a894-4b30e25c9e40)
[AssetRipper.log](https://github.com/AssetRipper/AssetRipper/files/12503674/AssetRipper.log)

**Comment by ds5678** [comment]
Maybe an exceptional was thrown in async code.

**Comment by Zbuffers** [comment]
Well,
even more strange:
all Images (Texture2D) were exported as .assets... ? (never seen that before)
note: I tried with a debug compile: same results.
Could it be that texture2D are converted to png during the  'Editor Format Conversion' phase ?
(it's in that that phase that the import 'seems' to be stuck), maybe I need to wait even more ? (I let it import for like 20+ minutes) before export last time in with debug build).

Note: the game has an enormous amount of bundles (Adressables): 5941 !!!
Maybe a relation ?

**Comment by ds5678** [comment]
> all Images (Texture2D) were exported as .assets

Are you sure these aren't Sprites? I don't know how textures could get exported the way you describe.

> Could it be that texture2D are converted to png during the 'Editor Format Conversion' phase ?

This processor has nothing to do with textures nor sprites.

> it's in that that phase that the import 'seems' to be stuck

Probably an animation-related slowdown

> the game has an enormous amount of bundles (Adressables): 5941 !!!

Larger games make performance problems more obvious.

**Comment by Zbuffers** [comment]
Yes I'm sure they aren't sprites, tha's the 1st thing i checked.

YAML FileID is  2800000 and 
it says Texture2D.

I have deleted all the exported project, will re-export without streaming assets and show you an example tomorow.

**Comment by Zbuffers** [comment]
Just tried to load WITHOUT streaming assets, and behaviour is normal. (load and export are fine) and Textures2D are indeed exported as Png.

I will try tonight to load WITH streaming assets, and let it run untill tomorow morning, to see if the load process comes to an end.

**Comment by Zbuffers** [comment]
Hi,
So it appears I am just too impatient.
The load process eventually finished after 3 hours...
the used RAM was very high (around 20 Gigs !!!)
And i started export.
It crashed after a wile because of an outOfMemory exception during export, but thats another issue, and not directely related to assetRipper. (was using more than 30Gigs of RAM).
The exported files are OK, no Texture2D .assets.

here is the log (should you be interested)
I'll try a last time wihout the GUI, just to check.
Sorry for bringing up this issue.
You can close it.
Z

[AssetRipper.log](https://github.com/AssetRipper/AssetRipper/files/12505353/AssetRipper.log)

**Comment by ds5678** [comment]
Thanks for the update. Is all well, now? What was up with the yaml textures?

**Comment by Zbuffers** [comment]
Hi,
Yes, all is well now. Using command did reduce the used RAM.

I did howether manage to reproduce the yaml texture export 'thing' ;)

It happened when i did not wait for the full load of the game, and started export too soon.

attached is an example of such yaml texture2D 
Also the log of the export (I chose shader decompile option so it would error out rapidly)

[UnityLogo.asset.txt](https://github.com/AssetRipper/AssetRipper/files/12506544/UnityLogo.asset.txt)
[AssetRipper.log](https://github.com/AssetRipper/AssetRipper/files/12506548/AssetRipper.log)
